### ISSUE TITLE
Feature request: "proper last played (with import)"

### ISSUE THREAD

**Issue by Friedfisch** [sandiz/rs-manager#27]
Hi there,
im tracking what i played for like 2-3? month so far  (doing snapshots of rocksniffer every 10?sec) - goal is to get a "last played" list on any track (lead/rythmn/alt/bass). so i have like 30k small json files but no time to write something to evaluate those :) - or put them in a (proper) database. 

Imho a nice addition to keep track and not play the same stuff all the time?

**Comment by JustinAiken** [comment]
Rocksmith itself saves a last played at in the profile... would that work instead of rocksniffer stuffs?

**Comment by Friedfisch** [comment]
No, RS does not keep track of time and day after some time - maybe for privacy i guess. And you can not cross-track sort it.

**Comment by sandiz** [comment]
hi @Friedfisch , thanks for the suggestion

> No, RS does not keep track of time and day after some time

i checked the dates associated with each arrangement in my profile and it seems to be tracking fine (learn-a-song)
```
cat /tmp/rs_player_data.json | jq '.Stats.Songs | with_entries(.value |= .DateLAS) 
| with_entries(select(.value != null)) | to_entries | sort_by(.value) | from_entries '
```
```
{
  "22769E2A77D5773A1BE50569F827901A": "2016-10-09 12:57:03",
  "174196B71E3930F6D3E5E935F2744E4D": "2016-10-09 13:29:35",
  "omitted for brevity" : "...", 
  "0EA163638702ED44C5665F9F04641CD0": "2018-10-18 14:58:02",
  "27468EC4C8219D2B3E21247A0976AC1B": "2018-10-19 14:35:06",
  "A8288088329D95105C275A8A37B89C97": "2018-10-19 14:40:16", 
  "B13D6A311A185EE871AFC47935002504": "2018-10-21 18:54:49", /* whisky lullaby - brad paisley pack */
  "54680F83BC2984C9FEC9AC27ADF99FA8": "2018-10-21 19:12:49"  /* whisky lullaby - brad paisley pack */
}
```
I have 910 records going back all the way to 2016. Do you know whats the cutoff for when its not tracking for you ?

> Imho a nice addition to keep track

yea, i want to add "recently played" too, but DateLAS in player profile seems to be most granular (its tracked on an arrangement level) so far...

the [data](https://gist.github.com/sandiz/535ebe97d65039ffb004fb3e19551e2c) sent by RockSniffer's is on a song id level, it doesn't tell me which arrangement you played. (though i think it can be added easily)

Here's what i am thinking:
v1. add DateLas support 
v2. add path support to RockSniffer + track that data in rs-manager (provided v1 data is insufficient)

Thoughts ?

**Comment by Firekorn** [comment]
The cutoff is only in display when sorting within RS but everything is indeed stored properly.

When sorting by Last Played in RS, they are only category for "Today", "Yesterday", "This week" and "Older" see : https://imgur.com/a/HhLyVo6

So i don't think the data in the profile is wrong, just that RS doesn't really use it more than for some relatively limited category.

**Comment by JustinAiken** [comment]
> So if i figure out what to actually run i can see all my profile data already?

Yes, your `_PRFLDB` file is just encrypted JSON - if you decrypt it, it has all kinds of useful info.  Everything is just keyed off the arrangement_id though, so you have to have those in a list to reference.

For example, `8C73667F0A0A601972496DBEFC42CF8B` is Bad Religion - Sorrow (Lead)

And in the big json blob, `json["Songs"]["8C73667F0A0A601972496DBEFC42CF8B"]["TimeStamp"]` will give you the last time you play it in Learn a Song.

**Comment by sandiz** [comment]
fixed in [`518b15b`](https://github.com/sandiz/rs-manager/commit/518b15b3970e140b844a6f842bc9a944608339c5), will be in the next release

**Comment by sandiz** [comment]
fixed in release [v1.1.0](https://github.com/sandiz/rs-manager/releases/tag/v1.1.0)
